### DISCUSSION TITLE
Graviton runtime

### DISCUSSION THREAD

**Discussion by kaihendry** [aws/serverless-application-model#2676] [Questions] [answered]
Hi, it's been awhile and perhaps this is better supported in `AWS::Serverless::Function`

[    Runtime: provided.al2
](https://github.com/kaihendry/ltabus/blob/a841621731ed61b6f1c0ee4a2e1a95270318e5a4/template.yml#L62)

How do I make sure I'm running the latest [Graviton 3](https://aws.amazon.com/ec2/graviton/)?

**Comment by hoffa** [comment, marked as answer]
AWS Lambda function only support Graviton2 currently by adding `arm64` to `Architectures`, see e.g. https://github.com/aws/aws-graviton-getting-started/blob/main/aws-lambda/README.md

**Reply by kaihendry** [reply]
So I don't need [Runtime: provided.al2](https://github.com/kaihendry/ltabus/blob/a841621731ed61b6f1c0ee4a2e1a95270318e5a4/template.yml#L62) anymore?

**Reply by hoffa** [reply]
Lambda functions using [.zip file archives](https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-package.html#gettingstarted-package-zip) need both. `Architecture` specifies the processor architecture, while `Runtime` specifies the software execution environment. You can see the list of other runtimes that support ARM (Graviton2) [here](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html). For Lambda [container images](https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-package.html#gettingstarted-package-images) `Runtime` isn't necessary as the container specifies the execution environment.

**Reply by kaihendry** [reply]
It would appear my https://github.com/kaihendry/ltabus/blob/a841621731ed61b6f1c0ee4a2e1a95270318e5a4/template.yml cannot be improved after reading https://github.com/aws/aws-graviton-getting-started/blob/main/aws-lambda/README.md
